tests/src/test/java/org/teavm/classlib/java/lang/TestObjectAsync.java

```diff
@@ -0,0 +1,42 @@
+/*
+ *  Copyright 2017 Alexey Andreev.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.teavm.classlib.java.lang;
+
+public class TestObjectAsync extends Object implements Runnable {
+    private int counter;
+    
+    public static Runnable r;
+
+    public TestObjectAsync() {
+        try {
+            Thread.sleep(100);
```

Comment:
> You don't need this line:

Ok; I was just mirroring the previous test. I have now changed it to use a class literal as you suggested.

> Is it really necessary to wait 100 milliseconds? Why not Thread.sleep(1)? Why not Thread.yield()?

True, Thread.sleep(1) is sufficient (now committed). Thread.yield() however does not trigger failure. The implementation of Thread.yield() appears to do nothing unless it is called at least 30 times:

```
   public static void yield() {
        TThread currentThread = currentThread();
        if (++currentThread.yieldCount < 30) {
            return;
        }
```

I suspect that many of the other tests, which do use Thread.yield(), would be more thorough if they instead used Thread.sleep(1).